src/hooks/useCredentialApi.ts

```diff
@@ -1,44 +1,165 @@
+  const { t } = useTranslation();
+
+  const apiCall = useCallback(
+    (ids: CredentialType['id'][]): Promise<AxiosResponse<ApiDeleteCredentialSuccessType>> =>
+      axios.post(`${process.env.REACT_APP_CREDENTIALS_SERVICE_BULK_DELETE}`, { ids }),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (
+      response: AxiosResponse<ApiDeleteCredentialSuccessType>,
+      updatedCredentials: CredentialType[]
+    ) => {
+      const { data } = response;
+
+      const deletedNames = data?.deleted
+        ?.map(id => updatedCredentials.find(cred => id === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const skippedNames = data?.skipped
+        ?.map(({ credential }) => updatedCredentials.find(cred => credential === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const missingNames = data?.missing
+        ?.map(id => updatedCredentials.find(cred => id === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const missingCredsMsg = `The following credentials could not be found: ${missingNames}`;
+
+      // All credentials were deleted successfully
+      if (data?.deleted?.length === updatedCredentials.length) {
+        const successMessage = t('toast-notifications.description', {
+          context: 'deleted-credential',
+          count: updatedCredentials.length,
+          name: updatedCredentials.map(({ name }) => name).join(', ')
+        });
+        onAddAlert({
+          title: successMessage,
+          variant: 'success',
+          id: helpers.generateId()
+        });
+        return;
+      }
+
+      // No credentials deleted, either all skipped or all missing
+      if (
+        data?.skipped?.length === updatedCredentials.length ||
+        data?.missing?.length === updatedCredentials.length
+      ) {
+        if (data?.skipped?.length === updatedCredentials.length) {
+          const errorMessage = t('toast-notifications.description', {
+            context: 'skipped-credential',
+            name: skippedNames,
+            count: data?.skipped?.length
+          });
+          onAddAlert({
+            title: errorMessage,
+            variant: 'danger',
+            id: helpers.generateId()
+          });
+        } else {
+          console.log(missingCredsMsg);
+        }
+        return;
+      }
+
+      // Some credentials deleted, some skipped or missing
+      if (data?.deleted?.length && data?.skipped?.length) {
+        const successMessage = t('toast-notifications.description', {
+          context: 'deleted-credentials-skipped-credentials',
+          deleted_names: deletedNames,
+          skipped_names: skippedNames,
+          count: data?.skipped?.length
+        });
+        onAddAlert({
+          title: successMessage,
+          variant: 'warning',
+          id: helpers.generateId()
+        });
+        if (data?.missing?.length) {
+          console.log(missingCredsMsg);
+          return;
+        }
+      }
```

Comment:
required: we have a unit test that is failing at displaying `success` messages on deletion, specifically, 
```
exports[`useDeleteCredentialApi should handle success while attempting to delete single and multiple credentials: deleteCredentials, success 1`] = []
```
This should be populated with messages...

**Snapshots are tricky**, they have be double checked manually the first few times to make sure the data is coming through. In this case your `success` messages aren't actually firing based on the logic you have

Recommendation
- Let's reset the logic, it appears we are working significantly harder than we need to. 
 
   Since our logic is concerned with things that aren't deleted let's refocus it on that. Let's try "process of elimination" and reverse the order so "success" is the last thing that happens. 
  
   What that means is we'll be limiting our conditions to prove "its not a success" since what we have currently is a bunch of messaging around that concept already, it's less conditions, less prone to developer error

```
      // No credentials deleted, either all skipped or all missing - Exact/Specific checks here are fine, let's do this first the OR gives us options. It's parallel to checking for "all deleted" but in the other direction
      if (
        data?.skipped?.length === updatedCredentials.length ||
        data?.missing?.length === updatedCredentials.length
      ) {
       ...
        onAddAlert({
          ...
        });

        if (data?.missing?.length) {
          console.log(missingCredsMsg);
        }
        return;
      }

      // Some credentials deleted, some skipped or missing - Truthy OR check instead, we don't really need to care beyond the existence of `skipped` or `missing` to fire the alerts. By their existence it means it didn't go according to plan
      if (data?.skipped?.length || data?.missing?.length) {
        ...
        onAddAlert({
          ...
        });

        if (data?.missing?.length) {
          console.log(missingCredsMsg);
        }
        return;
      }

     // All credentials were deleted successfully - **No conditional check necessary**. You can add one if you want, but it's unnecessary since we've eliminated all the skipped and missing before we got to this point
       ...
      onAddAlert({
        ...
      });
```

**OPTIONAL**
If you really wanted to take this even further... you started the concept of applying `missingCredsMsg` in the `console.log`... what's to stop you from doing that to the `onAddAlert` locale string, you could take that a step further and have a single `onAddAlert` at the end. the limitations could be seen as readability with the use of an `else if`... but it highlights that you can take even further... I see an additional optimization but I'll let you figure it out, the hint is ... you're doing string logic we could rely even more on the i18next translation/t function and just because "context" has a `property` doesn't mean it'll be used by the the translation/t function 

```
const message: AlertProps = {
        title: t('toast-notifications.description', {
          context: 'deleted-credential',
          count: updatedCredentials.length,
          name: updatedCredentials.map(({ name }) => name).join(', ')
        }),
        variant: 'success',
        id: helpers.generateId()
      };

// No credentials deleted, either all skipped or all missing
      if (
        data?.skipped?.length === updatedCredentials.length ||
        data?.missing?.length === updatedCredentials.length
      ) {
        message.title = t('toast-notifications.description', {
          context: 'skipped-credential',
          name: skippedNames,
          count: data?.skipped?.length
        });
        message.variant = 'danger';
      } else if (data?.skipped?.length || data?.missing?.length) {
        message.title = t('toast-notifications.description', {
          context: 'deleted-credentials-skipped-credentials',
          deleted_names: deletedNames,
          skipped_names: skippedNames,
          count: data?.skipped?.length
        });
        message.variant = 'warning';
      }

      // All credentials were deleted successfully
      onAddAlert(message);

      if (data?.missing?.length) {
          console.log(`The following credentials could not be found: ${missingNames}`);
      }

      return;
```